File: 구/SongheeLee/2606.py
import sys
input = sys.stdin.readline

def dfs(start):
    for i in graph[start]:
        visited[start] = True
        if not visited[i]:              # if visited[node] == False:
            dfs(i)

if __name__ == "__main__":
    node = int(input())
    line = int(input())

    graph = [[] for _ in range(node+1)]

    for _ in range(line):
        u, v = map(int, input().split())
        graph[u].append(v)
        graph[v].append(u)

    # for i in range(node+1):
    #     graph[i].sort()

    visited = [False]*(node+1)

    dfs(1)

    count = 0

    for i in visited:
        if i == True:
            count += 1
        
    print(count-1)

File: 구/YoungwooKim/2606.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let input = fs.readFileSync(filePath).toString().trim().split("\n");

const solution = (graph) => {
  let queue = [1];
  let count = 0;
  while (queue.length) {
    let v = queue.shift();
    if (visited[v]) continue;
    visited[v] = true;
    count++;
    graph[v].forEach((e) => {
      if (!queue.includes(e)) queue.push(e);
    });
  }
  console.log(count - 1);
};

const N = parseInt(input.shift());
const M = parseInt(input.shift());
const visited = Array.from({ length: N + 1 }, () => false);
const graph = Array.from({ length: N + 1 }, () => []);
// const visited = Array(N + 1).fill(false);
// const graph = Array(N + 1).fill([]);

for (let i = 0; i < M; i++) {
  const [a, b] = input[i].split(" ").map(Number);
  graph[a].push(b);
  graph[b].push(a);
}

solution(graph);
